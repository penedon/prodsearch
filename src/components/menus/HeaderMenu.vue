<template>
    <div class="outer-headerMenu">
        <div class="headerMenu"
             :class="{docked: docked}">
            <Transition name="fade" mode="out-in">
                <h1 v-if="!docked">PROD SEARCH</h1>
            </Transition>
            <Transition name="fade" mode="out-in">
                <h3 v-if="docked">PS</h3>
            </Transition>
            <div class="transitionSearchBox">
                <SearchBlock v-model:query="query"
                             :styleTransparent="docked"/>
            </div>
            
            <LocalizationMenu />
        </div>
        <Transition name="fade" mode="out-in">
            <div    v-if="docked"
                    class="headerMenuBar"></div>
        </Transition>
    </div>
</template>

<script>
import LocalizationMenu from '@/components/menus/LocalizationMenu.vue'
import SearchBlock from '@/components/blocks/SearchBlock.vue'

export default {
    emits: ['update:query'],
    components: {
        LocalizationMenu,
        SearchBlock,
    },
    props: {
        query: String,
        docked: Boolean
    },
    watch: {
        query(newValue) {
            this.$emit('update:query', newValue)
        }
    }
        
}
</script>

<style scoped>
    .outer-headerMenu {
        position: relative;
        width: 100%;
        display: inline-block;
        min-height: 52px;
    }
    .headerMenu {
        position: relative;
        width: 100%;
        z-index: 10;
    }
    .headerMenu.docked {
        height: 52px;
        position: fixed;
        background-color: var(--var-primary-color);
    }
    .headerMenuBar {
        width: 100%;
        height: 52px;
        top: 0;
        
        position: absolute;
        background-color: var(--var-primary-color);
        z-index: 1;
        /* box-shadow: 0 0px 10px black; */
    }

    h1 {
        position: absolute;
        top:200px;
        left: 50%;
        transform: translateX(-50%);
        /* text-shadow: 0 1px 1px #ccc; */
        color: var(--var-primary-color);
        font-size: 3em;
    }

    h3 {
        color: white;
        float: left;
        padding: 10px 15px 9px 15px;
        border-right: 2px solid var(--var-primary-color-light)
    }

    .transitionSearchBox {
        position: fixed;
        transform: translate(-50%, -50%);
        top:50%;
        left: 50%;
        width: 768px;
        height: 52px;
    }

    .headerMenu.docked .transitionSearchBox {
        /* position: fixed */
        top: 0 !important;
        left: 70px;
        transform: translate(0, 0);
        width: calc(100% - 200px);
    }


    .fade-enter-active, .fade-leave-active {
        transition: opacity 0.3s ease;
    }
    .fade-enter, .fade-enter-to, .fade-leave-from {
        opacity: 1;
    }
    .fade-leave, .fade-enter-from, .fade-leave-to {
        opacity: 0;
    }

    @media only screen and (max-width: 768px) {
        .headerMenu:not(.docked) .transitionSearchBox {
            width: 90%;
            /* position: relative;
            transform: translate(0,0);
            top:0; left: 0;
            margin: 0 auto; */
        }
        .headerMenu:not(.docked) h1 {
            /* position: relative; */
            /* top: 100px; */
            /* left: 0; */
            /* transform: translateX(0); */
        }
    }
</style>