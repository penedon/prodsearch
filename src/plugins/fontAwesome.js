/* import the fontawesome core */
import { library } from '@fortawesome/fontawesome-svg-core'

/* import specific icons */
import { faArrowUpRightFromSquare, faCloudMoon, faCloudSun, faCog, faEnvelope, faMagnifyingGlass, faTimes, faUser } from '@fortawesome/free-solid-svg-icons'
import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome'
import { faLinkedin, faLinkedinIn } from '@fortawesome/free-brands-svg-icons'

library.add(
    faUser,
    faEnvelope,
    faLinkedin,
    faLinkedinIn,
    faMagnifyingGlass,
    faCloudSun,
    faCloudMoon,
    faCog,
    faTimes,
    faArrowUpRightFromSquare
)

export {
    FontAwesomeIcon
}