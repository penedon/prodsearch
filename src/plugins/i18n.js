import { createI18n } from 'vue-i18n'

import en from '@/locales/en'
import es from '@/locales/es'
import pt from '@/locales/pt'

const messages = {
    'en': en,
    'es': es,
    'pt': pt
};

const i18n = new createI18n({
  locale: 'en', // set locale
  fallbackLocale: 'es', // set fallback locale
  globalInjection: true,
  messages, // set locale messages
})

export {
    i18n
};