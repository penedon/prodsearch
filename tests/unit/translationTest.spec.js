import { shallowMount } from '@vue/test-utils'
import { i18n } from '@/plugins/i18n'
import HeaderMenu from '@/components/menus/HeaderMenu'


describe('test translation', () => {
    function initWrapper(component) {
        return shallowMount(component, {
          global: {
            plugins: [i18n]
          }
        })
    }
    let wrapper = initWrapper(HeaderMenu);

    test('Load Component in English', () => {
        wrapper = initWrapper(HeaderMenu);
        expect(wrapper.element).toMatchSnapshot();
    });

    test('translates component to Spanish', () => {
        wrapper = initWrapper(HeaderMenu);
        wrapper.componentVM.$i18n.locale = 'es'
        expect(wrapper.element).toMatchSnapshot();
    });

    test('translates component to Portuguese', () => {
      wrapper = initWrapper(HeaderMenu);
      wrapper.componentVM.$i18n.locale = 'pt'
      expect(wrapper.element).toMatchSnapshot();
  });

})
