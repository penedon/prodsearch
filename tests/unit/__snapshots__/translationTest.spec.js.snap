// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test translation Load Component in English 1`] = `
<div
  class="outer-headerMenu"
>
  <div
    class="headerMenu"
  >
    <transition-stub>
      <h1>
        PROD SEARCH
      </h1>
    </transition-stub>
    <transition-stub>
      <!--v-if-->
    </transition-stub>
    <div
      class="transitionSearchBox"
    >
      <search-block-stub
        styletransparent="false"
      />
    </div>
    <localization-menu-stub />
  </div>
  <transition-stub>
    <!--v-if-->
  </transition-stub>
</div>
`;

exports[`test translation translates component to Portuguese 1`] = `
<div
  class="outer-headerMenu"
>
  <div
    class="headerMenu"
  >
    <transition-stub>
      <h1>
        PROD SEARCH
      </h1>
    </transition-stub>
    <transition-stub>
      <!--v-if-->
    </transition-stub>
    <div
      class="transitionSearchBox"
    >
      <search-block-stub
        styletransparent="false"
      />
    </div>
    <localization-menu-stub />
  </div>
  <transition-stub>
    <!--v-if-->
  </transition-stub>
</div>
`;

exports[`test translation translates component to Spanish 1`] = `
<div
  class="outer-headerMenu"
>
  <div
    class="headerMenu"
  >
    <transition-stub>
      <h1>
        PROD SEARCH
      </h1>
    </transition-stub>
    <transition-stub>
      <!--v-if-->
    </transition-stub>
    <div
      class="transitionSearchBox"
    >
      <search-block-stub
        styletransparent="false"
      />
    </div>
    <localization-menu-stub />
  </div>
  <transition-stub>
    <!--v-if-->
  </transition-stub>
</div>
`;
